allauth/account/forms.py

```diff
@@ -283,22 +285,29 @@ def __init__(self, *args, **kwargs):
             self.fields['email'].required = False
             self.fields['email'].widget.is_required = False
             if self.username_required:
-                field_order = ['username', 'email']
-                if app_settings.SIGNUP_EMAIL_ENTER_TWICE:
-                    field_order.append('email2')
-
-        # Merge our email and username fields in if they are not
-        # currently in the order.  This is to allow others to
-        # re-arrange email and username if they desire.  Go in reverse
-        # so that we make sure the inserted items are always
-        # prepended.
-        for field in reversed(field_order):
-            if field not in merged_field_order:
-                merged_field_order.insert(0, field)
-        set_form_field_order(self, merged_field_order)
+                default_field_order = [
+                    'username',
+                    'email',
+                    'email2',  # ignored when not present
+                    'password1',
+                    'password2'  # ignored when not present
+                ]
+
         if not self.username_required:
             del self.fields["username"]
 
+        if hasattr(self, 'field_order'):
+            set_form_field_order(self, self.field_order)
+        else:
+            default_field_order = [
```

Comment:
Setting the `default_field_order` here overrides the logic above, I think this needs to be dropped?